### PR TITLE
Remove TSP App and Public API flags

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

There is no Public API endpoint anymore so I've removed the flags around serving that endpoint.

There is also no TSP app anymore so I've gone ahead and removed the remaining bits so we can disable route 53 DNS.

## Reviewer Notes

Are there things that I need to be aware of before removing the TSP app? I think its fairly trivial to bring back and it reduces our attack surface area on the app. But if I've missed a product decision I want to be aware of it.

## Setup

No setup needed, just watch the tests pass.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/169020283) for this change

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ services:
   milmove_migrate:
     depends_on:
       - database
-    image: 923914045601.dkr.ecr.us-west-2.amazonaws.com/app-migrations:git-branch-placeholder_branch_name
+    image: 923914045601.dkr.ecr.us-west-2.amazonaws.com/app-migrations:git-branch-master
```
Comment (mr337) [inline docker-compose.yml]: Should this be reverted? Not seeing connection between this and the removal of TSP.

**Comment by chrisgilmerproj** [comment]
There's a failing server test that is unrelated to the changes. I'm trying to track it down but so far haven't been able to fix it. These same tests pass elsewhere and revolve around testing TLS connection with the fake certs we provide.

File: Makefile [hunk 2]
```diff
@@ -553,7 +551,6 @@ endif
 db_test_start: ## Start Test DB
 ifndef CIRCLECI
 	brew services stop postgresql 2> /dev/null || true
-endif
 	@echo "Starting the ${DB_DOCKER_CONTAINER_TEST} docker database container..."
```
Comment (lynzt) [inline Makefile]: so this is done via circle now?

File: Makefile [hunk 3]
```diff
@@ -396,9 +391,9 @@ server_test_standalone: ## Run server unit tests with no deps
 ifndef CIRCLECI
 	DB_NAME=$(DB_NAME_TEST) DB_PORT=$(DB_PORT_TEST) go test -count 1 -short $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/ | grep -v mocks)
 else
-	# Limit the maximum number of tests to run in parallel to 8 for CircleCI due to memory constraints.
+	# Limit the maximum number of tests to run in parallel for CircleCI due to memory constraints.
 	# Add verbose (-v) so go-junit-report can parse it for CircleCI results
-	DB_NAME=$(DB_NAME_TEST) DB_PORT=$(DB_PORT_TEST) go test -v -parallel 8 -count 1 -short $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/ | grep -v mocks)
+	DB_NAME=$(DB_NAME_TEST) DB_PORT=$(DB_PORT_TEST) go test -v -parallel 4 -count 1 -short $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/ | grep -v mocks)
```
[the remark this answers is not in the document]
Comment (chrisgilmerproj) [reply Makefile]: Trying to turn down the resource load on the containers. There's some magic number here that I don't understand yet but felt like 8 was too high.

File: Makefile (hunk 2 above)
Comment (chrisgilmerproj) [reply Makefile]: Circle always handled the DBs for the server tests. But not for the integration tests. I think when we moved to docker-compose for the integration tests I forgot to clean this one up. But previously this step was running and failing and then continuing on for all server tests. So I went ahead and cut out the extra time it was spending on this part of the code.

Comment (lynzt) [reply Makefile]: cool, thanks for that context

**Pull request merged**